crates/uv-distribution-types/src/index.rs

```diff
@@ -160,14 +160,28 @@ impl Index {
     /// For indexes with a `/simple` endpoint, this is simply the URL with the final segment
     /// removed. This is useful, e.g., for credential propagation to other endpoints on the index.
     pub fn root_url(&self) -> Option<Url> {
-        let segments = self.raw_url().path_segments()?;
-        let last = segments.last()?;
+        let mut segments = self.raw_url().path_segments()?;
+        let mut last = segments.next_back()?;
```

Comment:
I think there's a `pop_if_empty` that you might be able to use here?

Reply:
(Honestly, I didn't read the diff closely enough to know if it helps for sure, but it kind of looked like it from a glance.)

Reply:
Not in the immutable version, but yes I can use it later. One sec...

Reply:
It's a bit cleaner, thanks.